src/step.rs

```diff
@@ -89,15 +90,15 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 let dest_layout = self.type_layout(dest_ty)?;
 
                 match *dest_layout {
-                    Layout::General { discr, ref variants, .. } => {
-                        let discr_size = discr.size().bytes();
-                        let discr_offset = variants[variant_index].offsets[0].bytes();
+                    Layout::General { discr, .. } => {
+                        // FIXME: I (oli-obk) think we need to check the
+                        // `dest_ty` for the variant's discriminant and write
+                        // instead of the variant index
+                        // We don't have any tests actually going through these lines
```

Comment:
I think on `-Copt-level=3` we actually get `SetDiscriminant` calls. I'm not sure if we actually need to do what your `FIXME` says, since (I believe) variant indexes always match discriminants for `Layout::General` right now. But it would be less brittle code.